### PR TITLE
Check HPACK Huffman code strictly

### PR DESCRIPTION
**Pull request by maskit**
This fixes issues checked with these h2spec tests below.

5.2. String Literal Representation
1: Sends a Huffman-encoded string literal representation with padding longer than 7 bits
2: Sends a Huffman-encoded string literal representation padded by zero

### CODE REVIEW

**Comment by maskit** [comment]
@zwoop This is nice to have because it's not a bug fix. Marked as a 7.1 candidate but there's no necessity.

File: proxy/http2/HuffmanCodec.cc [hunk 1]
```diff
@@ -154,26 +154,41 @@ huffman_decode(char *dst_start, const uint8_t *src, uint32_t src_len)
...
     if (shift) {
       --shift;
     } else {
       shift = 7;
       ++src;
       --src_len;
+      ++byte_boundary_crossed;
     }
+    if (byte_boundary_crossed > 3) {
+      return -1;
```
Comment (masaori335) [inline proxy/http2/HuffmanCodec.cc]: Why not use `HPACK_ERROR_COMPRESSION_ERROR`?

Comment (maskit) [inline proxy/http2/HuffmanCodec.cc]: Because that's in HPACK.h, so we shouldn't use it here.

Comment (masaori335) [inline proxy/http2/HuffmanCodec.cc]: Ah, here is HuffmanCodec. Got it.

File: proxy/http2/HuffmanCodec.cc [hunk 2]
```diff
@@ -151,29 +151,44 @@ hpack_huffman_fin()
...
     }
     if (shift) {
       --shift;
     } else {
       shift = 7;
       ++src;
       --src_len;
+      ++byte_boundary_crossed;
     }
+    if (byte_boundary_crossed > 3) {
```
Comment (masaori335) [inline proxy/http2/HuffmanCodec.cc]: This is checking length of encoded data, right? It looks like some Huffman code has 3 byte boundaries. Is this should be 4? ( e.g. `|11111111|11111111|11111110|0010         fffffe2  [28]`)

Comment (masaori335) [inline proxy/http2/HuffmanCodec.cc]: Never mind, I thought `>= 3` :p

**Comment by masaori335** [comment]
[approve ci autest]

**Comment by zwoop** [comment]
I've put this on Docs / CI for testing.

**Comment by zwoop** [comment]
Should we land this?

**Comment by zwoop** [comment]
@maskit This does not cherry-pick cleanly to 7.1.x branch, can either of you make a PR against 7.1.x please?

**Comment by maskit** [comment]
@zwoop Made it (#2200). Actually it was able to cherry-pick cleanly. It just has two commits.

**Pull request merged**